shared/protocol/src/main/java/io/pravega/shared/protocol/netty/WireCommands.java

```diff
@@ -2539,4 +2558,38 @@ public void release() {
          */
         abstract void releaseInternal();
     }
+
+    /**
+     * Convenience class to encapsulate the contents of an attribute update when several should be serialized in the same
+     * WireCommand.
+     */
+    @Data
+    public static final class ConditionalAttributeUpdate {
+        public static final int LENGTH = 4 * Long.BYTES + 1; // UUID (2 longs) + oldValue + newValue + updateType (1 byte)
+
+        private final UUID attributeId;
+        private final byte attributeUpdateType;
```

Comment:
We need to add at least some static final constants with what values to put here. The Controller won't know the semantics of this, nor does it have access to the Segment store's `AttributeUpdateType.` I suggest we add 2 constants whose values match exactly the ones in `AttributeUpdateType`. For now, we can have `Replace` and `ReplaceIfEquals`. We  can add the other ones later if needed.

Reply:
And we'd have to somehow validate that the values encoded are correct.

Reply:
Done, and added tests in `PravegaRequestProcessorTests` to validate the values.